flink-table-store-core/src/main/java/org/apache/flink/table/store/file/operation/AbstractFileStoreScan.java

```diff
@@ -206,7 +206,22 @@ public Plan plan() {
                             "Unknown value kind " + entry.kind().name());
             }
         }
-        List<ManifestEntry> files = new ArrayList<>(map.values());
+        List<ManifestEntry> files = new ArrayList<>();
+        for (ManifestEntry file : map.values()) {
+            if (checkNumOfBuckets) {
+                Preconditions.checkState(
```

Comment:
It may be better to throw the (TableException?) exception manually, which may be a common exception encountered by users.